src/main/java/org/alfresco/repo/search/adaptor/lucene/QueryConstants.java

```diff
@@ -187,7 +187,11 @@
 
     public static final String FIELD_SOLR_UNIT_OF_TIME_HOUR_SUFFIX = "_unit_of_time_hour";
 
-    public static final String FIELD_SOLR_UNIT_OF_TIME_DAY_SUFFIX = "_unit_of_time_day";
+    public static final String FIELD_SOLR_UNIT_OF_TIME_DAY_SUFFIX = "_unit_of_time_day_of_month";
```

Comment:
I suspect it is better to change the name of the variable to "FIELD_SOLR_UNIT_OF_TIME_DAY_OF_MONTH_SUFFIX"

Reply:
Actually looked up the examples from SQL and it seems that DAY is used as DAY_OF _MONTH, so it's all fine.